lambda/src/main/scala/pricemigrationengine/migrations/GuardianWeekly2025Migration.scala

```diff
@@ -172,22 +172,36 @@ object GuardianWeekly2025Migration {
   // - amendmentOrderPayload is used in the Amendment handler
   // ------------------------------------------------
 
+  def logValue[T](label: String)(value: T): T = {
+    println(s"$label: $value")
+    value
+  }
+
   def priceData(
       subscription: ZuoraSubscription,
       invoiceList: ZuoraInvoiceList,
       account: ZuoraAccount
   ): Either[DataExtractionFailure, PriceData] = {
     val priceDataOpt: Option[PriceData] = for {
-      ratePlan <- SI2025RateplanFromSubAndInvoices.determineRatePlan(subscription, invoiceList)
-      currency <- SI2025Extractions.determineCurrency(ratePlan)
-      oldPrice = SI2025Extractions.determineOldPrice(ratePlan)
-      localisation <- SubscriptionLocalisation.determineSubscriptionLocalisation(subscription, invoiceList, account)
-      billingPeriod <- SI2025Extractions.determineBillingPeriod(ratePlan)
+      _ <- Some(()).map(logValue("initialization"))
+      ratePlan <- SI2025RateplanFromSubAndInvoices
+        .determineRatePlan(subscription, invoiceList)
+        .map(logValue("ratePlan"))
```

Comment:
not important, but `tap` might be more idiomatic - https://zio.dev/reference/core/zio/#tapping

Reply:
Oh. Nice one. I saw that being used in the ZIO code, but I didn't know it was basic Scala 🙏